river/evaluate/progressive_validation.py

```diff
@@ -48,7 +51,10 @@ def _progressive_validation(
         y_pred = preds.pop(i)
         if y_pred != {} and y_pred is not None:
             metric.update(y_true=y, y_pred=y_pred)
-        model.learn_one(x=x, y=y)
+        if model._supervised:
+            model.learn_one(x=x, y=y)
+        else:
+            model.learn_one(x=x)
```

Comment:
This merits some discussion. What about if we want to do `model.learn_one(x)` before the prediction/scoring? That's actually desirable in some cases. Do you have any thoughts on this?

Reply:
It could be that I'm wrong here, but isn't a progressive validation a test-then-train evaluation?
If we still want to have the `model.learn_one(x)` before the prediction/scoring, I would avert the problem at a higher level. When calling the `run` method of a `Track` we could pass a new argument for a different validation procedure.

Reply:
Agreed. I think I'm convinced :)